__tests__/cookies-page.test.js

```diff
@@ -97,6 +110,37 @@ describe('Cookies page', () => {
     )
   })
 
+  it('injects Google Tag Manager script if user accepts cookies', async () => {
+    // Click 'Yes' and submit
+    await $radioYes.click()
+    await $buttonSave.click()
+
+    expect(
+      page.$('script[src*="www.googletagmanager.com"]')
+    ).resolves.toBeTruthy()
+  })
+
+  it('does not inject Google Tag Manager script if user rejects cookies', async () => {
+    await $radioNo.click()
+    await $buttonSave.click()
+
+    expect(
+      page.$('script[src*="www.googletagmanager.com"]')
+    ).resolves.toBeNull()
+  })
+
+  it('immediately disables analytics if user rejects cookies', async () => {
+    await $radioNo.click()
+    await $buttonSave.click()
+
+    expect(
+      page.evaluate(() => window['ga-disable-G-8F2EMQL51V'])
+    ).resolves.toEqual(true)
+    expect(
+      page.evaluate(() => window['ga-disable-G-GHT8W0QGD9'])
```

Comment:
Are these 'fixed' codes that we know won't change? Is there any documentation around them?

Reply:
They're the IDs of our Google Analytics properties, so they'll only change if we set up new analytics properties (hopefully not anytime soon now we migrated to GA4).

In the actual code for the analytics, we have [constants for their last part with a little explanation](https://github.com/alphagov/govuk-design-system/blob/main/src/javascripts/components/cookie-functions.mjs#L21-L22) as well [as some comments when we set `ga-disable-...`](https://github.com/alphagov/govuk-design-system/blob/main/src/javascripts/components/cookie-functions.mjs#L21-L22). 

There are also [unit tests checking that the function managing the cookies sets or unsets them as necessary](https://github.com/alphagov/govuk-design-system/blob/main/src/javascripts/components/cookie-functions.mjs#L21-L22), but there was nothing checking that this was actually happening when interacting with the page, just that cookies were set.